docs/Configuration.md

```diff
@@ -0,0 +1,106 @@
+```json
+// appsettings.Development.json
+{
+  "Infrastructure": {
+    // Secret that should not be exposed
+    "DialogDbConnectionString": "TODO: Add to local secrets", 
+    "Altinn": {
+      // Non secret with a sensible value for local development
+      "BaseUri": "https://platform.tt02.altinn.no/"
+    }
+  }
+}
+```
+
+# 2. Normal configuration
+By "normal configuration" we mean non secrets or configuration that that does not pose a security risk when shared externally. This type of configuration can live directly in the git repository, which gives us benefits like version controlled configuration, and a single point of truth for each environments configuration. 
+
+Each runtime defines an `appsettings.{Environment}.json` for each target environment. This file should contain environment specific configuration. In addition to the specific environment configuration files there should also exist a `appsettings.json`. This file should contain cross cutting environment configuration. The `appsettings.{Environment}.json` configuration takes presence over the `appsettings.json` configuration, meaning - if the configuration key `foo` exists in both files, the value from `appsettings.{Environment}.json` is used. 
+
+## 2.1. Word of caution
+Due to the implicit and cross cutting nature of `appsettings.json` **it should be used cautiously.** It's often better to let a runtime fail hard and crash due to no configuration, than using the wrong configuration value silently. Say for example that a `.well-known` authentication service endpoint is configured in `appsettings.json` because the same endpoint is used in `Development`, `Test` and `Stage`. However, the developer forgets to add it to `appsettings.Production.json` and thereby not overriding it for the `Production` environment. Thus `Production` is now using the same `.well-known` endpoint as the rest of the environments and thereby open to non-production tokens. It should produce a **big fat red critical error**, but instead it continues on like everything is fine. Therefore the `.well-known` endpoint, and many other configurations, does **not belong** in `appsettings.json`. **The absence of a configuration is often better than the wrong one silently working.** 
+
+## 2.2. How to
+Adding, removing or updating non secret configuration is as simple as modifying `appsettings.json` for cross cutting configuration or `appsettings.[Environment].json` for environment specific configuration. **Use caution when adding configuration to `appsettings.json`**, see [2.1. Word of caution](#21-word-of-caution). When adding a required configuration to `appsettings.[Environment].json`, remember to add the configuration for all the environment specific json files. GitHub actions will do the rest when merging to main.
+
+## 2.3. CI/CD
```

Comment:
Kanskje få inn noe her rundt bruken av labels per runtime for Azure App Configuration og hvordan dette håndteres med label filters i `AddAzureAppConfiguration`

Reply:
Jeg la på følgende i seksjon 2.3:
>Each runtime / presentation layer adds its own configuration keys to azure app configuration through `appsettings.json` and `appsettings.[Environment].json` with a runtime configuration prefix, indicating which runtime that owns that specific configuration value. An environment label will also be added to the configuration based on the target environment.

Er det ok?